dot/state/storage.go

```diff
@@ -107,6 +119,26 @@ func (s *StorageState) StoreTrie(ts *rtstorage.TrieState) error {
 	s.tries[root] = ts.Trie()
 	s.lock.Unlock()
 
+	if _, ok := s.pruner.(*fullNodePruner); header == nil && ok {
+		return fmt.Errorf("block cannot be empty for Full node pruner")
+	}
+
+	if header != nil {
+		insKeys, err := ts.GetInsertedNodeHashes()
```

Comment:
could you add a test case for these changes also?

Reply:
Done